pydatastructs/utils/misc_util.py

```diff
@@ -35,6 +36,41 @@ class TreeNode(Node):
         Optional, index of the right child node.
     """
 
+    __slots__ = ['key', 'data', 'left', 'right', 'is_root',
+                 'height', 'parent', 'size']
+
+    def __new__(cls, key, data):
+        obj = Node.__new__(cls)
+        obj.data, obj.key = data, key
+        obj.left, obj.right, obj.parent, obj.height, obj.size = \
+            None, None, None, 0, 1
+        obj.is_root = False
+        return obj
+
+    def __str__(self):
+        """
+        Used for printing.
+        """
+        return str((self.left, self.key, self.data, self.right))
+
+class RedBlackTreeNode(Node):
```

Comment:
`Node` -> `TreeNode`?

Reply:
No luck, still giving the same error.

Reply:
error is originating from `OneDimensionalArray.__setitem__()`

Reply:
I found the bug. will send the PR with its possible fix.